tests/e2e/Services/Sites/SitesCustomServerTest.php

```diff
@@ -2393,4 +2393,41 @@ public function testInvalidSSRSource(): void
 
         $this->cleanupSite($siteId);
     }
+
+    public function testPermanentRedirect(): void
+    {
+        $siteId = $this->setupSite([
+            'siteId' => ID::unique(),
+            'name' => 'Sub project site',
+            'framework' => 'other',
+            'buildRuntime' => 'node-22',
+            'outputDirectory' => './'
+        ]);
+        $this->assertNotEmpty($siteId);
+
+        $site = $this->getSite($siteId);
+        $this->assertEquals('200', $site['headers']['status-code']);
+        $this->assertArrayHasKey('adapter', $site['body']);
+
+        $domain = $this->setupSiteDomain($siteId);
+        $this->assertNotEmpty($domain);
+
+        $deploymentId = $this->setupDeployment($siteId, [
+            'code' => $this->packageSite('subProjects'),
```

Comment:
Let's rename to sub-directories, it better describes what we are testing.

Also `sub-directories` not `subDirectories` for folder name